### PR TITLE
[MODULAR] Add Some Species Previews

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
I was gonna do the whole lot, but I got distracted, and I'd rather not let this get lost in my 30+ local branches!

Fixes Akula, Dwarf, Ghoul, Hemophage, Humanoid, Insect and Mammal previews!

image
image

## How This Contributes To The Skyrat Roleplay Experience
It doesn't, it just makes picking a race more pretty!

## Changelog

:cl:
fix: Fixed previews for Akula, Dwarf, Ghoul, Hemophage, Humanoid, Insect and Mammal races!
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm [hunk 1]
```diff
@@ -35,3 +35,13 @@
 
 /datum/species/humanoid/get_species_lore()
 	return list("Make sure you fill out your own custom species lore!")
+
+/datum/species/humanoid/prepare_human_for_preview(mob/living/carbon/human/human)
+
+/datum/species/humanoid/prepare_human_for_preview(mob/living/carbon/human/human)
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm]: ?

Comment (RimiNosha) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm]: Uhhh, good question!

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm [hunk 2]
```diff
@@ -84,3 +84,8 @@
     "Historically, Hemophages have caused great societal strife through their very existence. Many have reported dread on having someone reveal they require blood to survive, worse on learning they have been undead, espiecally in 'superstitious' communities. In many places they occupy a sort of second class, unable to live normal lives due to their condition being a sort of skeleton in their closet. Some can actually be found in slaughterhouses or the agricultural industry, gaining easy access to a large supply of animal blood to feed their eternal thirst.",
     "Others find their way into mostly-vampiric communities, turning others into their own kind; though, the virus can only transmit to hosts that are incredibly low on blood, taking advantage of their reduced immune system efficiency and higher rate of blood creation to be able to survive the initial few days within their host.",
     "\"What the fuck does any of this mean?\" - Doctor Micheals, reading their CentCom report about the new 'hires'.")
+
+/datum/species/hemophage/prepare_human_for_preview(mob/living/carbon/human/human)
+	human.skin_tone = "albino"
+	human.update_mutant_bodyparts(TRUE)
+	human.update_body(TRUE)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: If you're going to update them, at least give them some hair.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm [hunk 3]
```diff
@@ -35,3 +35,11 @@
...
 /datum/species/humanoid/get_species_lore()
 	return list("Make sure you fill out your own custom species lore!")
+
+/datum/species/humanoid/prepare_human_for_preview(mob/living/carbon/human/human)
+	var/main_color = skintone2hex("caucasian1")
+	human.dna.features["mcolor"] = main_color
+	human.dna.features["mcolor2"] = main_color
+	human.dna.features["mcolor3"] = main_color
+	human.update_mutant_bodyparts(TRUE)
+	human.update_body(TRUE)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm]: Could probably go for some hair too.

That being said, isn't humanoid intended to be used for species that _don't_ have skintones that match humans?

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/mammal.dm [hunk 4]
```diff
@@ -109,3 +109,16 @@
...
+/datum/species/mammal/prepare_human_for_preview(mob/living/carbon/human/human)
+	var/main_color = "#333333"
+	var/secondary_color = "#b8b8b8"
+	human.dna.features["mcolor"] = main_color
+	human.dna.features["mcolor2"] = secondary_color
+	human.dna.features["mcolor3"] = secondary_color
+	human.dna.species.mutant_bodyparts["ears"] = list(MUTANT_INDEX_NAME = "Husky", MUTANT_INDEX_COLOR_LIST = list(main_color, secondary_color, "#464646"))
+	human.dna.species.mutant_bodyparts["snout"] = list(MUTANT_INDEX_NAME = "Husky", MUTANT_INDEX_COLOR_LIST = list(main_color, secondary_color, secondary_color))
+	human.dna.species.mutant_bodyparts["tail"] = list(MUTANT_INDEX_NAME = "Husky", MUTANT_INDEX_COLOR_LIST = list(main_color, "#4D4D4D", secondary_color))
+	human.dna
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/mammal.dm]: This should be removed, it's what's failing your Windows Build.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/akula.dm [hunk 5]
```diff
@@ -81,3 +81,14 @@
...
 	return list(placeholder_lore)
+
+/datum/species/akula/prepare_human_for_preview(mob/living/carbon/human/human)
+	var/main_color = "#394b66"
+	var/secondary_color = "#818b9b"
+	human.dna.features["mcolor"] = main_color
+	human.dna.features["mcolor2"] = secondary_color
+	human.dna.features["mcolor3"] = secondary_color
+	human.dna.species.mutant_bodyparts["tail"] = list(MUTANT_INDEX_NAME = "Shark", MUTANT_INDEX_COLOR_LIST = list(main_color, secondary_color, secondary_color))
+	human.dna.species.mutant_bodyparts["snout"] = list(MUTANT_INDEX_NAME = "hShark", MUTANT_INDEX_COLOR_LIST = list(main_color, secondary_color, secondary_color))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/akula.dm:90]: Don't Akulas also have like, a fin or something, on their head? That portrait looks really odd at the moment, any way to make it look less like an egg would be good in my books.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
If you need help with the Screenshot tests, hit me up. But I wouldn't suggest updating them until we're satisfied with the looks of the previews.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/mammal.dm (hunk 4 above)
Comment (RimiNosha) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/mammal.dm]: Huh, I have no idea how I left that one in-

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm (hunk 3 above)
Comment (ghost) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/humanoid.dm]: Yea, I was gonna say the same, could we use like, a greenish or reddish or something, make it look like an Orc or Devil of some sort? Would help emphasize it as humanOID and not just human

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm (hunk 2 above)
Comment (ghost) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: I vote giving the Hemophage the most over-the-top and outdated hairstyle you can, make them look like a proper vampire 😆

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/akula.dm (hunk 5 above)
Comment (ghost) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/akula.dm:90]: looked into it, found some pretty cursed knowledge: they DONT, not much. They have ears, but-

`/datum/sprite_accessory/ears/mutant/akula/sergal`

They're SERGAL ears. `/mutant/akula` is just an empty basetype that defines the recommended species, `/mutant/akula/sergal` ears are the only ones with a texture that are technically akula ears

Do with that what you will

**Comment by GoldenAlpharex** [comment]
Hit me up when you address those changes!

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm [hunk 6]
```diff
@@ -87,5 +87,7 @@
 
 /datum/species/hemophage/prepare_human_for_preview(mob/living/carbon/human/human)
 	human.skin_tone = "albino"
+	human.hair_color = "#000000"
+	human.hairstyle = "Long Sideparted"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: Just for aesthetic reasons, I think we should probably try and go with a hairstyle that covers the face a bit less. I can go and look for one and hit you up on Discord with a suggestion for a replacement, if you'd like.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm (hunk 6 above)
Comment (RimiNosha) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: I have no clue what hairstyles really exist that'd fit the bill- someone did mention Pompadour, though I haven't checked it yet.

**Review by GoldenAlpharex — COMMENTED** [review]
Looks good, the last thing you have to do is fix the screenshots tests, hit me up on Discord if you need any help with that.

**Pull request merged**